orttraining/orttraining/core/graph/gradient_builder.cc

```diff
@@ -194,7 +194,8 @@ IMPLEMENT_GRADIENT_BUILDER(GetMatMulGradient) {
     }
   };
 
-  if (A_has_shape && B_has_shape && Y_has_shape) {
+  if (A_has_shape && B_has_shape && Y_has_shape &&
+      A_shape.size() >= 2 && B_shape.size() >= 2) {
```

Comment:
Matmul schema only allow inputs with dim>=2. 
I thought this is an implicit prerequisite? Are you seeing inputs with less then 2 dim?

Reply:
I saw the discussion on team's channel, and understand now this is a torch bug/"feature". 
This fix is harmless.

Reply:
Out MatMul allows vectors.

---
In reply to: [571201490](https://github.com/microsoft/onnxruntime/pull/6581#discussion_r571201490) [](ancestors = 571201490)